src/collectors/vm/prometheus_vm_memory_collector.erl

```diff
@@ -27,8 +27,9 @@ register() ->
 register(Registry) ->
   ok = prometheus_registry:register_collector(Registry, ?MODULE).
 
-deregister(_) -> ok.
+deregister(_) -> true.
```

Comment:
what's the idea behind this? if I remember correctly deregiter called by prometheus_registry.clear and its return value is  basically ignored. 
in fact you just highlighted how deregister is flawed. You register collector using `register/0,1` but you can't deregister just this one collector and you shouldn't call `deregister/1` yourself! wtf??
OK that's because deregister is merely a cleanup routing. goh.
I'll rewrite this a little bit so you'll have to rebase :innocent: .